From f984f716f1fbe1bec41a9eb1c89fc3d60bfe2c17 Mon Sep 17 00:00:00 2001
From: Jason Gerecke <killertofu@gmail.com>
Date: Thu, 8 Aug 2024 12:05:42 -0700
Subject: [PATCH libwacom] data: Add stylus definition for Pro Pen 3E

The Pro Pen 3E is a version of the Pro Pen 3 which includes an eraser
and is compatible with the same devices. This patch also adds the
propen3 group to Movink, which should have had it.

Signed-off-by: Jason Gerecke <jason.gerecke@wacom.com>
(cherry picked from commit 59aaf9e6db27b088a7c37bc85e0954b1c1f968ae)
---
 data/cintiq-pro-17.tablet |  2 +-
 data/cintiq-pro-22.tablet |  2 +-
 data/cintiq-pro-27.tablet |  2 +-
 data/libwacom.stylus      | 21 ++++++++++++++++++++-
 data/movink-13.tablet     |  2 +-
 5 files changed, 24 insertions(+), 5 deletions(-)

diff --git a/data/cintiq-pro-17.tablet b/data/cintiq-pro-17.tablet
index d8f55dcd8078..77a258acb2f1 100644
--- a/data/cintiq-pro-17.tablet
+++ b/data/cintiq-pro-17.tablet
@@ -33,7 +33,7 @@ DeviceMatch=usb:056a:03c4
 Width=15
 Height=8
 Layout=cintiq-pro-17.svg
-Styli=@cintiqpro2022;@mobilestudio;@propengen2;
+Styli=@propen3;@mobilestudio;@propengen2;
 IntegratedIn=Display
 
 [Features]
diff --git a/data/cintiq-pro-22.tablet b/data/cintiq-pro-22.tablet
index befdb0a7a532..c9825e9682ad 100644
--- a/data/cintiq-pro-22.tablet
+++ b/data/cintiq-pro-22.tablet
@@ -33,7 +33,7 @@ DeviceMatch=usb:056a:03d0
 Width=18
 Height=10
 Layout=cintiq-pro-22.svg
-Styli=@cintiqpro2022;@mobilestudio;@propengen2;
+Styli=@propen3;@mobilestudio;@propengen2;
 IntegratedIn=Display
 
 [Features]
diff --git a/data/cintiq-pro-27.tablet b/data/cintiq-pro-27.tablet
index db0c5dbc20b7..fd7b883b9fff 100644
--- a/data/cintiq-pro-27.tablet
+++ b/data/cintiq-pro-27.tablet
@@ -33,7 +33,7 @@ DeviceMatch=usb:056a:03c0
 Width=24
 Height=13
 Layout=cintiq-pro-27.svg
-Styli=@cintiqpro2022;@mobilestudio;@propengen2;
+Styli=@propen3;@mobilestudio;@propengen2;
 IntegratedIn=Display
 
 [Features]
diff --git a/data/libwacom.stylus b/data/libwacom.stylus
index 60f2ffb7eda0..1426d285c410 100644
--- a/data/libwacom.stylus
+++ b/data/libwacom.stylus
@@ -294,7 +294,16 @@ Type=General
 [0x200]
 # Cintiq Pro 2022
 Name=Pro Pen 3
-Group=cintiqpro2022
+Group=propen3
+Buttons=3
+Axes=Tilt;Pressure;Distance;
+Type=General
+
+[0x202]
+# Wacom Movink 13
+Name=Pro Pen 3E
+Group=propen3
+PairedStylusIds=0x20a;
 Buttons=3
 Axes=Tilt;Pressure;Distance;
 Type=General
@@ -460,6 +469,16 @@ Buttons=2
 Axes=Tilt;Pressure;Distance;
 Type=General
 
+[0x20A]
+# Wacom Movink 13
+Name=Pro Pen 3E Eraser
+Group=propen3
+PairedStylusIds=0x202;
+EraserType=Invert
+Buttons=3
+Axes=Tilt;Pressure;Distance;
+Type=General
+
 [0x85a]
 # Intuos2
 Name=Grip Pen Eraser
diff --git a/data/movink-13.tablet b/data/movink-13.tablet
index 40c634263a57..27943baefeb2 100644
--- a/data/movink-13.tablet
+++ b/data/movink-13.tablet
@@ -36,7 +36,7 @@ DeviceMatch=usb:056a:03f0
 Class=Cintiq
 Width=11
 Height=8
-Styli=@udpen;@mobilestudio;@propengen2;
+Styli=@udpen;@mobilestudio;@propengen2;@propen3;
 IntegratedIn=Display
 Layout=movink-13.svg
 
-- 
2.51.0

